File: Demos/CSharp/09. Addendas/Adenda/Kuehne.cs
using HyperSoft.Ejemplo.Data;
using HyperSoft.ElectronicDocumentLibrary.Kuehne.Addenda;

namespace HyperSoft.Ejemplo.Adendas
{
  internal static partial class Adenda
  {
    internal static bool Kuehne(out string fileName)
    {
      //En este método se cargan los datos de la factura.
      Cfdi40.CargarDatosCompleto(electronicDocument);

      Kuehne addenda = new HyperSoft.ElectronicDocumentLibrary.Kuehne.Addenda.Kuehne();

      addenda.Data.PurchaseOrder.Value = "aaaaaaaaaaaaaa";
      addenda.Data.FileNumberGl.Value = "00000000000000";
      addenda.Data.BranchCentre.Value = "aaaaaaa";
      addenda.Data.TransportRef.Value = "aaaaaaa";

      electronicDocument.Data.Addendas.Add(addenda);

      return Save("Addenda_Kuehne.xml", out fileName);
    }
  }
}

File: Demos/Visual Basic .Net/09. Addendas/Adenda/Kuehne.vb
Imports HyperSoft.Ejemplo.Data
Imports HyperSoft.ElectronicDocumentLibrary.Kuehne.Addenda

Friend NotInheritable Partial Class Adenda
	
	Friend Shared Function Kuehne(ByRef fileName As String) As Boolean
		'En este método se cargan los datos de la factura.
		Cfdi40.CargarDatosCompleto(electronicDocument)

		Dim addenda As Kuehne = New HyperSoft.ElectronicDocumentLibrary.Kuehne.Addenda.Kuehne()

		addenda.Data.PurchaseOrder.Value = "aaaaaaaaaaaaaa"
		addenda.Data.FileNumberGl.Value = "00000000000000"
		addenda.Data.BranchCentre.Value = "aaaaaaa"
		addenda.Data.TransportRef.Value = "aaaaaaa"

		electronicDocument.Data.Addendas.Add(addenda)

		Return Save("Addenda_Kuehne.xml", fileName)
	End Function

End Class